src/blob.rs

```diff
@@ -266,6 +269,34 @@ impl<'a> BlobObject<'a> {
             _ => (stem, format!(".{}", ext).to_lowercase()),
         }
     }
+
+    /// Checks whether a name is a valid blob name.
+    ///
+    /// This is slightly less strict than stanitise_name, presumably
+    /// someone already created a file with such a name so we just
+    /// ensure it's not actually a path in disguise is actually utf-8.
+    ///
+    /// # Return value
+    ///
+    /// If this is a valid name a reference to the [str]
+    /// representation is returned.  Otherwise a [BlobError] with
+    /// [BlobErrorKind::WrongName] is returned.
+    fn is_blob_name<'b>(name: impl AsRef<OsStr>) -> bool {
```

Comment:
nitpick: i'd name this `is_acceptable_blob_name` to make sure it's answering a theoretical question, not a question about anything in the blob store.